src/Build/Graph/GraphBuilder.cs

```diff
@@ -510,9 +513,32 @@ private ParsedProject ParseProject(ConfigurationMetadata configurationMetadata)
                 throw new InvalidOperationException(ResourceUtilities.GetResourceString("NullReferenceFromProjectInstanceFactory"));
             }
 
-            var graphNode = new ProjectGraphNode(projectInstance);
+            if (enableDynamicPlatformResolution)
+            {
+                string referencedProjectPlatform = projectInstance.GetPropertyValue(PlatformMetadataName);
+                string projectReferencePlatformsMetadata = projectInstance.GetPropertyValue(PlatformsMetadataName);
+                string projectReferenceLookupTableMetadata = projectInstance.GetPropertyValue(PlatformLookupTableMetadataName);
+                String projectPath = projectInstance.FullPath;
+                String platformLookupTable;
+                globalProperties.TryGetValue(PlatformLookupTableMetadataName, out platformLookupTable);
+                String CurrentProjectPlatform;
+                globalProperties.TryGetValue(PlatformMetadataName, out CurrentProjectPlatform);
+                var SelectedPlatform = PlatformNegotiation.GetNearestPlatform(referencedProjectPlatform, projectReferencePlatformsMetadata, projectReferenceLookupTableMetadata, platformLookupTable, projectPath, CurrentProjectPlatform);
+
+                if (!string.IsNullOrEmpty(SelectedPlatform))
+                {
+                    globalProperties["platform"] = SelectedPlatform;
+                }
+
+                projectInstance = _projectInstanceFactory(
```

Comment:
Shouldn't this only re-evaluate if the `Platform` global property was set? Looks like this should be moved into the if statement above.

Reply:
it needs to be reevaluated anyways because we need to put its platformlookuptable in its global props in order to passdown.

Reply:
I also added in all the tests from GetCompatiblePlatform_Tests to GetCompatiblePlatformgraph_Tests and they all pass. so the behavior is the same here

Reply:
^ Correcting myself. The second evaluation is _always_ needed no matter what, because under `EnableDynamicPlatformResolution=true` you always start with the "empty eval". Next, imagine platform negotiation fails and you didn't add `Platform` to the global props. **You still need to do the second eval** with whatever global props may have been passed in. Re-resolving :)